sdk/iot/core/tests/cmocka/main.c

```diff
@@ -24,9 +27,27 @@ void az_iot_sas_token_generate_sas_token_overflow_fails(void** state);
 void az_iot_sas_token_generate_succeeds(void** state);
 void az_iot_sas_token_generate_with_keyname_succeeds(void** state);
 
+/*
+ * Telemetry Unit Tests
+ */
+void az_iot_hub_client_telemetry_publish_topic_get_NULL_client_fails(void** state);
```

Comment:
To me, it feels like these are too verbose and are not actually providing a lot of extra value. We should have a reduced set of tests that can cover all blocks and definitely not cover 2^n permutations of every single possible call an app can make (especially since the code is not that complex).
We could also either reduce the number of functions of provide include files with both the definitions and the lists.

Reply:
I think I have just enough tests here. One test to check for each invalid param. Make sure that each optional param is covered in the successful case and in the case where there isn't enough room. And that if we have some combination of options each of those are covered (since size checks are done for each possible combo)